code/modules/shuttle/custom_shuttle.dm

```diff
@@ -0,0 +1,246 @@
+	src.add_fingerprint(usr)
+	if(!allowed(usr))
+		to_chat(usr, "<span class='danger'>Access denied.</span>")
+		return
+
+	if(href_list["calculate"])
+		calculateStats()
+		ui_interact(usr)
+		return
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return
+	if(M.launch_status == ENDGAME_LAUNCHED)
+		return
+	if(href_list["setloc"])
+		SetTargetLocation(href_list["setloc"])
+		ui_interact(usr)
+		return
+	else if(href_list["fly"])
+		Fly()
+		ui_interact(usr)
+		return
+
+/obj/machinery/computer/custom_shuttle/proc/calculateDistance(var/obj/docking_port/stationary/port)
+	var/deltaX = port.x - x
+	var/deltaY = port.y - y
+	var/deltaZ = (port.z - z) * Z_DIST
+	return sqrt(deltaX * deltaX + deltaY * deltaY + deltaZ * deltaZ)
+
+/obj/machinery/computer/custom_shuttle/proc/linkShuttle(var/new_id)
+	shuttleId = new_id
+	possible_destinations = "whiteship_home;shuttle[new_id]_custom"
+
+/obj/machinery/computer/custom_shuttle/proc/calculateStats(var/useFuel = FALSE, var/dist = 0, var/ignore_cooldown = FALSE)
+	if(!ignore_cooldown && stat_calc_cooldown >= world.time)
+		to_chat(usr, "<span>You are using this too fast, please slow down</span>")
+		return
+	stat_calc_cooldown = world.time + CALCULATE_STATS_COOLDOWN
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return FALSE
+	//Reset data
+	calculated_mass = 0
+	calculated_dforce = 0
+	calculated_speed = 0
+	calculated_engine_count = 0
+	calculated_consumption = 0
+	calculated_cooldown = 0
+	calculated_fuel_less_thrusters = 0
+	calculated_non_operational_thrusters = 0
+	//Calculate all the data
+	var/list/areas = M.shuttle_areas
+	for(var/shuttleArea in areas)
+		for(var/each in shuttleArea)
+			var/atom/atom = each
+			if(!atom)
+				continue
+			//Items weight too little to add on 100kg
+			if(!istype(atom, /obj/item))
+				continue
+			calculated_mass ++
+			if(!istype(atom, /obj/machinery/shuttle/engine))
+				continue
+			var/obj/machinery/shuttle/engine/E = atom
+			E.check_setup()
+			if(!E.thruster_active)	//Skipover thrusters with no valid heater
+				calculated_non_operational_thrusters ++
+				continue
+			if(E.attached_heater)
+				if(!E.attached_heater.hasFuel(dist * E.fuel_use) && useFuel)
+					calculated_fuel_less_thrusters ++
+					continue
+			calculated_engine_count++
+			calculated_dforce += E.thrust
+			calculated_consumption += E.fuel_use
+			calculated_cooldown = max(calculated_cooldown, E.cooldown)
+	//This should really be accelleration, but its a 2d spessman game so who cares
+	if(calculated_mass == 0)
+		return FALSE
+	calculated_speed = (calculated_dforce*1000) / (calculated_mass*100)
+	return TRUE
+
+/obj/machinery/computer/custom_shuttle/proc/consumeFuel(var/dist)
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return FALSE
+	//Calculate all the data
+	for(var/obj/machinery/shuttle/engine/shuttle_machine in GLOB.custom_shuttle_machines)
+		shuttle_machine.check_setup()
```

Comment:
You need to check that the shuttle_machine is on the same shuttle as the computer.

Reply:
Fixed